ui/lib/CreateStream.tsx

```diff
@@ -25,32 +26,67 @@ export interface CreateStreamProps {
 }
 
 const CreateStream = (props: CreateStreamProps) => {
+  const [balance, setBalance] = useState<number | null>(null);
+  const [tokenBalance, setTokenBalance] = useState<number | null>(null);
+  const [gasPrice, setGasPrice] = useState<number | null>(null);
+  const { data, isError, isLoading } = useFeeData();
   const { address } = useAccount();
+
+  // Set balances for native and stream tokens
+  useEffect(() => {
+    if (address) {
+      (async () => {
+        // Fetch and set native balance as integer
+        const balanceData = await fetchBalance({ address });
+        if (balanceData && balanceData.formatted !== undefined) {
+          setBalance(Number(balanceData.formatted));
+        }
+        // Fetch and set token balance as integer
+        const tokenBalanceData = await fetchBalance({
+          address,
+          token: props.tokenAddress,
+        });
+        if (tokenBalanceData && tokenBalanceData.formatted !== undefined) {
+          setTokenBalance(Number(tokenBalanceData.formatted));
+        }
+      })();
+    }
+  }, [address]);
+
+  // Get gas price
+  useEffect(() => {
+    if (data && data.formatted && data.formatted.gasPrice !== undefined) {
+      setGasPrice(Number(data.formatted.gasPrice));
+    } else {
+      setGasPrice(null);
+    }
+  }, [data, isError, isLoading]);
```

Comment:
You're only using `data` here though?

Reply:
Good catch!

Reply:
Was useful for testing but no need in prod